src/components/Modal/AddLiquidity.vue

```diff
@@ -602,6 +618,22 @@ export default {
     },
     formatBalance(balanceString, tokenDecimals) {
       return normalizeBalance(balanceString, tokenDecimals);
+    },
+    async canAddLiquidity() {
+      // Can always add liquidity to a shared pool
+      // Can add liquidity to a smart pool unless there is a whitelist (and this address isn't on it)
+      if (
+        this.bPool.metadata.crp &&
+        this.bPool.metadata.rights.canWhitelistLPs
+      ) {
+        // Need to check if this address is on the LP whitelist
+        return await canProvideLiquidity(
+          this.bPool.metadata.controller,
+          this.web3.account
```

Comment:
This should be changed as discussed
```suggestion
          this.web3.dsProxyAddress
```

Reply:
Last commit did this